src/bpftrace.cpp

```diff
@@ -724,8 +726,86 @@ void perf_event_lost(void *cb_cookie, uint64_t lost)
   bpftrace->out_->lost_events(lost);
 }
 
-std::unique_ptr<AttachedProbe> BPFtrace::attach_probe(Probe &probe, const BpfOrc &bpforc)
+std::vector<std::unique_ptr<AttachedProbe>> BPFtrace::attach_usdt_probe(
+    Probe &probe,
+    std::tuple<uint8_t *, uintptr_t> func,
+    int pid,
+    bool file_activation)
+{
+  std::vector<std::unique_ptr<AttachedProbe>> ret;
+
+  // File activation works by scanning through /proc/*/maps and seeing
+  // which processes have the target executable in their address space
+  // with execute permission. If found, we will try to attach to each
+  // process we find.
+  if (file_activation && probe.path.size())
+  {
+    glob_t globbuf;
+    if (::glob("/proc/[0-9]*/maps", GLOB_NOSORT, nullptr, &globbuf))
+      throw std::runtime_error("failed to glob");
+
+    const char *p;
+    if (!(p = realpath(probe.path.c_str(), nullptr)))
+    {
+      std::cerr << "Failed to resolve " << probe.path << std::endl;
+      return ret;
+    }
+    std::string resolved(p);
+
+    for (size_t i = 0; i < globbuf.gl_pathc; ++i)
+    {
+      std::string path(globbuf.gl_pathv[i]);
+      std::ifstream file(path);
+      if (file.fail())
+        throw std::runtime_error("failed to open " + path);
+
+      std::string line;
+      while (std::getline(file, line))
+      {
+        if (line.find(resolved) == std::string::npos)
+          continue;
+
+        auto parts = split_string(line, ' ');
+        if (parts.at(1).find('x') == std::string::npos)
+          continue;
+
+        // Remove `/proc/` prefix
+        std::string pid_str(globbuf.gl_pathv[i] + 6);
+        // Remove `/maps` suffix
+        for (int i = 0; i < 5; ++i)
+          pid_str.pop_back();
+
+        int pid_parsed;
+        try
+        {
+          pid_parsed = std::stoi(pid_str);
+        }
+        catch (const std::exception &ex)
+        {
+          throw std::runtime_error("failed to parse pid=" + pid_str);
+        }
+
+        ret.emplace_back(
+            std::make_unique<AttachedProbe>(probe, func, pid_parsed));
```

Comment:
```suggestion
            std::make_unique<AttachedProbe>(probe, func, pid_parsed));
            break;
```

Don't think we have to process the rest of the file now